pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -430,20 +465,28 @@ func (w *wlReconciler) nominateAndSynchronizeWorkers(ctx context.Context, group
 	log.V(3).Info("Nominate and Synchronize Worker Clusters")
 	var nominatedWorkers []string
 
-	if w.dispatcherName == config.MultiKueueDispatcherModeAllAtOnce {
-		for workerName := range group.remotes {
-			nominatedWorkers = append(nominatedWorkers, workerName)
-		}
-		if group.local.Status.ClusterName == nil && !equality.Semantic.DeepEqual(group.local.Status.NominatedClusterNames, nominatedWorkers) {
-			group.local.Status.NominatedClusterNames = nominatedWorkers
-			if err := workload.ApplyAdmissionStatus(ctx, w.client, group.local, true, w.clock); err != nil {
-				log.V(2).Error(err, "Failed to patch nominated clusters", "workload", klog.KObj(group.local))
-				return reconcile.Result{}, err
+	// For elastic workloads, retrieve the remote cluster where the original workload was scheduled.
+	// For now, new workload slices will continue to be assigned to the same cluster.
+	// In the future, we may introduce more nuanced remote workload propagation policies,
+	// supporting preferred or required placement constraints.
+	if clusterName := workload.ClusterName(group.local); group.IsElasticWorkload() && clusterName != "" {
+		nominatedWorkers = []string{clusterName}
+	} else {
+		if w.dispatcherName == config.MultiKueueDispatcherModeAllAtOnce {
```

Comment:
Related comment, but I don't like the dispatcher code needs to be changed. As mentioned in https://github.com/kubernetes-sigs/kueue/pull/6445/files#r2360609035 this will generate firction to require developers of external dispatchers to do some custom code. In case of ElasticJobs, at least for MVP, we assume pinning to the same cluster. So, I see no point going via the nomination phase, and complicating the code of disaptchers. cc @tenzen-y

Reply:
Let's continue in https://github.com/kubernetes-sigs/kueue/pull/6445#discussion_r2360589742 to avoid duplication in comment.